examples/data-objects/prosemirror/src/prosemirror.tsx

```diff
@@ -76,7 +77,7 @@ export class ProseMirror extends EventEmitter implements IFluidLoadable, IProvid
 	}
 
 	public text: SharedString | undefined;
-	private root: ISharedMap | undefined;
+	private root: (ISharedMap & ISharedObject) | undefined;
```

Comment:
this feels like defeats the purpose of this change, no? I.e. I'd think that our examples should demonstrate the usage patterns, and if we have to add back ISharedObject, then we sort of lost.

Reply:
Note that this is an implementation of a custom data object: that is not something at all supported in the public/declarative API so the fact that users of the encapsulated API have to refer to these encapsulated API specific types does not defeat the purpose of this change. I think it would be way less confusing for developers working on the declarative API if we had any examples of it in our examples, but we don't since our example setup is tied to the encapsulated API.

If we add back ISharedObject, then we have to add back IFluidRuntimeDatastore, delta manager, IChannel, etc. Basically the whole encapsulated API would be pulled into the declarative API. That would defeat the purpose of a ton of other change.

Reply:
We do have declarative API examples. They intentionally live outside the main repo in https://github.com/microsoft/FluidExamples